gorfc/gorfc.go

```diff
@@ -1216,3 +1221,41 @@ func (conn *Connection) Call(goFuncName string, params interface{}) (result map[
 	}
 	return wrapResult(funcDesc, funcCont, C.RFC_IMPORT, conn.rstrip)
 }
+
+func (conn *Connection) rfcCancel(goFuncName string) error {
+	var errorInfo C.RFC_ERROR_INFO
+	rc := C.RfcCancel(conn.handle, &errorInfo)
+	if rc != C.RFC_OK {
+		return rfcError(errorInfo, "Could not invoke function \"%v\"", goFuncName)
+	}
+	conn.alive = false
+	return nil
+}
+
+// Call calls the given function with the given parameters and wraps the results returned.
+func (conn *Connection) Call(goFuncName string, params interface{}) (result map[string]interface{}, err error) {
+	return conn.CallContext(context.Background(), goFuncName, params)
+}
+
+// CallContext calls the given function with the given parameters and wraps the results returned.
+func (conn *Connection) CallContext(ctx context.Context, goFuncName string, params interface{}) (result map[string]interface{}, err error) {
+	if ctx.Done() == nil {
+		return conn.rfcInvoke(goFuncName, params)
+	}
+
+	done := make(chan interface{})
```

Comment:
Assuming idea is to notify about execution state, to avoids sending arbitrary values into a channel by making done a channel of type error. 

I think it will make it clearer and safer in case of concurrency issues, also, it's easy improvement, simpty:

```
    done := make(chan error)
    go func() {
        result, err = conn.rfcInvoke(goFuncName, params)
        done <- err
        close(done)
    }()
```
And few tweaks in `select case` will be needed.